### PR TITLE
Refactor/optional return

### PR DESCRIPTION
**Pull request by hnicke**
After screening how the api controllers have to handle calls to the service layer, I've wanted to change the service layers to always return the value wrapped in Optionals (java 8 feature) in case this value might be null. After the change, the api layer looks way cleaner.
I believe the usage of Optional will help us build more stable software.

Change:
- `csarService.getCsar(String): Csar` --> `csarService.getCsar(String): Optional<Csar>`
- `platformService.FindPlatformById(String): Platform` --> `platformService.FindPlatformById(String): Optional<Platform>`
- `csar.getTemplate()`: TServiceTemplate` --> `csar.getTemplate(): Optional<TServiceTemplate>`

Add:
- `csar.getTranformation(Platform): Optional<Transformation>`

### CODE REVIEW

File: core/src/main/java/org/opentosca/toscana/core/transformation/execution/ExecutionTask.java [hunk 1]
```diff
@@ -72,7 +73,10 @@ private void serveArtifact() {
 
     private void transform() {
         try {
-            plugin.transform(new TransformationContext(transformation, csarContentDir, transformationRootDir));
+            TServiceTemplate template = transformation.getCsar().getTemplate().orElseThrow(() ->
+                new IllegalStateException("TServiceTemplate is null, will not start transformation."));
```
Comment (c-mueller) [inline core/src/main/java/org/opentosca/toscana/core/transformation/execution/ExecutionTask.java]: Maybe reword this to "TServiceTemplate of csar {} is null, could not start transformation"

Comment (hnicke) [reply core/src/main/java/org/opentosca/toscana/core/transformation/execution/ExecutionTask.java]: fixed
